language/stdlib/modules/gas_schedule.mvir

```diff
@@ -0,0 +1,324 @@
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 49, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 46, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 47, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 46, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 39, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 29, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 34, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 32, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 30, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 856, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 929, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 929, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 917, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 774, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 1119, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 29, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 41, storage: 1 }
+        );
+        Vector.push_back<Self.Cost>(
+          &mut instruction_table,
+          Cost { cpu: 10, storage: 1 }
+        );
+
+        move_to_sender<T>(T {
+            instruction_schedule: move(instruction_table),
+            native_schedule: Vector.empty<Self.Cost>(),
+        });
+        return;
+    }
+
+    public new_cost(cpu: u64, storage: u64): Self.Cost {
+      return Cost {cpu: move(cpu), storage: move(storage) };
+    }
+
+    // Add a cost to the cost table. Whether the cost is native or not is specified by the
+    // `is_instruction_cost` variable that is passed in.
+    // Note: This adds the instruction onto the end of the vector.
+    public add_cost(is_instruction_cost: bool, cost: Self.Cost) acquires T {
+        let gas_schedule: &mut Self.T;
+        assert(get_txn_sender() == 0xA550C18, 1);
```

Comment:
will we move to a capability story here? should this tie to the Libra Upgrade Protocol (formerly known as social transactions, formerly known as WriteSet, formerly known as FTVM :D)

Reply:
Maybe? I don't know how much a capability story would give us here other than allowing us to delegate the ability to add add a cost. 

In terms of tying in with the Libra Upgrad Protocol, we would need primitives/natives for determining the calling context in order to enforce such things no?